src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,675 @@
+  return FD_LAYOUT_FINI( l, scratch_align() );
+}
+
+FD_FN_CONST static inline void *
+mux_ctx( void * scratch ) {
+  return (void*)fd_ulong_align_up( (ulong)scratch, alignof(fd_repair_tile_ctx_t) );
+}
+typedef struct __attribute__((packed)) {
+  fd_eth_hdr_t eth[1];
+  fd_ip4_hdr_t ip4[1];
+  fd_udp_hdr_t udp[1];
+} eth_ip_udp_t;
+
+static inline void
+populate_packet_header_template( eth_ip_udp_t * pkt,
+                                 ulong          payload_sz,
+                                 uint           src_ip,
+                                 uchar const *  src_mac,
+                                 ushort         src_port ) {
+  memset( pkt->eth->dst, 0,       6UL );
+  memcpy( pkt->eth->src, src_mac, 6UL );
+  pkt->eth->net_type  = fd_ushort_bswap( FD_ETH_HDR_TYPE_IP );
+
+  pkt->ip4->verihl       = FD_IP4_VERIHL( 4U, 5U );
+  pkt->ip4->tos          = (uchar)0;
+  pkt->ip4->net_tot_len  = fd_ushort_bswap( (ushort)(payload_sz + sizeof(fd_ip4_hdr_t)+sizeof(fd_udp_hdr_t)) );
+  pkt->ip4->net_frag_off = fd_ushort_bswap( FD_IP4_HDR_FRAG_OFF_DF );
+  pkt->ip4->ttl          = (uchar)64;
+  pkt->ip4->protocol     = FD_IP4_HDR_PROTOCOL_UDP;
+  pkt->ip4->check        = 0U;
+  memcpy( pkt->ip4->saddr_c, &src_ip, 4UL );
+  memset( pkt->ip4->daddr_c, 0,       4UL ); /* varies by shred */
+
+  pkt->udp->net_sport = fd_ushort_bswap( src_port );
+  pkt->udp->net_dport = (ushort)0; /* varies by shred */
+  pkt->udp->net_len   = fd_ushort_bswap( (ushort)(payload_sz + sizeof(fd_udp_hdr_t)) );
+  pkt->udp->check     = (ushort)0;
+}
+
+static void FD_FN_UNUSED
+send_packet( fd_repair_tile_ctx_t * ctx,
+             uint    ip,
+             ushort  port,
+             uchar const * payload,
+             ulong   payload_sz,
+             ulong   tsorig ) {
+  uchar * packet = fd_chunk_to_laddr( ctx->net_out_mem, ctx->net_out_chunk );
+
+  eth_ip_udp_t * hdr = (eth_ip_udp_t *)packet;
+  uchar mac[6] = {0};
+  populate_packet_header_template( hdr, payload_sz, ctx->repair_my_intake_addr.addr, mac, ctx->repair_intake_listen_port );
+
+  hdr->udp->net_dport = port;
+
+  memcpy( hdr->eth->dst, mac, 6UL );
+  memcpy( hdr->ip4->daddr_c, &ip, 4UL );
+
+  // TODO: LML handle checksum correctly
+  hdr->ip4->check = fd_ip4_hdr_check( ( fd_ip4_hdr_t const *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4 ) );
+
+  ulong packet_sz = payload_sz + sizeof(eth_ip_udp_t);
+  fd_memcpy( packet+sizeof(eth_ip_udp_t), payload, payload_sz );
+
+  hdr->udp->check = fd_ip4_udp_check( *(uint *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4->saddr_c ), 
+                                      *(uint *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4->daddr_c ), 
+                                      (fd_udp_hdr_t const *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->udp ), 
+                                      packet + sizeof(eth_ip_udp_t) );
+
+  ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+  ulong sig = fd_disco_netmux_sig( ip, port, FD_NETMUX_SIG_MIN_HDR_SZ, DST_PROTO_OUTGOING, (ushort)0 );
```

Comment:
The args here don't seem to make sense, `FD_NETMUX_SIG_MIN_HDR_SZ` in not a valid dst ip address?